earth_enterprise/src/fusion/dbmanifest/dbmanifest.cpp

```diff
@@ -327,6 +332,64 @@ void DbManifest::GetManifest(geFilePool &file_pool,
   GetXmlManifest(stream_manifest);
 }
 
+// Read POI file and extract the data file names from within it and then add
+// them to the manifest too
+void DbManifest::GetPoiDataFiles(const std::string& poi_file,
+                     std::vector<ManifestEntry>* stream_manifest,
+                     std::vector<ManifestEntry>* search_manifest)
+{
+  notify(NFY_DEBUG,
+        "Parsing POI file %s looking for data files", poi_file.c_str());
+  size_t stream_poi_file_idx = stream_manifest->size() - 1;
+  size_t search_poi_file_idx = search_manifest->size() - 1;
+  khxml::DOMLSParser *parser = CreateDOMParser();
+  if (parser) {
+    khxml::DOMDocument *doc = ReadDocument(parser, poi_file);
+    if (doc) {
+      try {
+        if (khxml::DOMElement *root = doc->getDocumentElement()) {
+          if (khxml::DOMElement* el_search_table = GetFirstNamedChild(root, "SearchTableValues")) {
+            std::vector<std::string> data_files;
+            FromElementWithChildName(el_search_table, "SearchDataFile", data_files);
+            if (data_files.empty()) {
+              notify(NFY_WARN,
+                    "No data files defined for POI file %s", poi_file.c_str());
+            }
+            for (uint i = 0; i < data_files.size(); ++i) {
+
+              const std::string& data_file = data_files[i];
+              
+              // poi data files have to be set aside until after AddDB processing is done
+              // so we hide them in a special location now within the manifest and will
+              // use them when we crate an 'upload' manifest during DBsync processing
+              if (stream_manifest != search_manifest) {
```

Comment:
I guess if you don't pass the full vector you're going to run into trouble here, but you could do the check in the outer function and pass a boolean or something.

Reply:
Actually I thought about that and what I'll do is pass in the pointers to the entries.  If the vector pointers point to the same vector so will the entry pointers point to the same entry.